graph/src/components/store.rs

```diff
@@ -1444,16 +1444,20 @@ pub enum EntityModification {
     /// Insert the entity
     Insert { key: EntityKey, data: Entity },
     /// Update the entity by overwriting it
-    Overwrite { key: EntityKey, data: Entity },
+    Overwrite {
+        key: EntityKey,
+        data: Entity,
+        prev_vid: Vid,
+    },
     /// Remove the entity
-    Remove { key: EntityKey },
+    Remove { key: EntityKey, prev_vid: Vid },
```

Comment:
The vid type is optional, but deep in the clamp query code it is treated as a required value. Is there a way we could already treat it as required here?

Reply:
Yes, I am not happy about that either; the issue is that when mappings create entities, we don't have a `vid` yet; I'll think about how to address that by refining the `EntityVersion` further (or, less excited about that, have the store handle `None` `vid`'s properly)

Reply:
Got it.

Reply:
I looked at this some more, and we'll have to deal with `vid` possibly being `None` in some form until we transact the changes. We can express the transition from '`None` is permissible' to '`None` is not allowed' differently, but I think it will always be a dynamic property and can't be enforced statically.

But AFAICT, what the store does is safe in the sense that a `None` in the wrong place will lead to a subgraph failure, not to corrupted data.